src/Microsoft.Azure.EventHubs.Processor/AzureBlobLease.cs

```diff
@@ -3,83 +3,68 @@
 
 namespace Microsoft.Azure.EventHubs.Processor
 {
-    using System;
-    using Newtonsoft.Json;
-    using WindowsAzure.Storage;
-    using WindowsAzure.Storage.Blob;
+	using System.Threading.Tasks;
+	using Newtonsoft.Json;
+	using WindowsAzure.Storage;
+	using WindowsAzure.Storage.Blob;
 
-    class AzureBlobLease : Lease
-    {
-        string offset = null;
+	class AzureBlobLease : Lease
+	{
+		// ctor needed for deserialization
+		internal AzureBlobLease()
+		{
+		}
 
-        // ctor needed for deserialization
-        internal AzureBlobLease()
-        {
-        }
+		internal AzureBlobLease(string partitionId, CloudBlockBlob blob) : base(partitionId)
+		{
+			this.Blob = blob;
+		}
 
-        internal AzureBlobLease(string partitionId, CloudBlockBlob blob)
-            : base(partitionId)
-        {
-            this.Blob = blob;
-        }
+		internal AzureBlobLease(AzureBlobLease source)
+			: base(source)
+		{
+			this.Offset = source.Offset;
+			this.SequenceNumber = source.SequenceNumber;
+			this.Blob = source.Blob;
+		}
 
-        internal AzureBlobLease(AzureBlobLease source)
-            : base(source)
-        {
-            this.Offset = source.Offset;
-            this.SequenceNumber = source.SequenceNumber;
-            this.Blob = source.Blob;
-        }
+		internal AzureBlobLease(AzureBlobLease source, CloudBlockBlob blob) : base(source)
+		{
+			this.Offset = source.Offset;
+			this.SequenceNumber = source.SequenceNumber;
+			this.Blob = blob;
+		}
 
-        internal AzureBlobLease(AzureBlobLease source, CloudBlockBlob blob)
-            : base(source)
-        {
-            this.Offset = source.Offset;
-            this.SequenceNumber = source.SequenceNumber;
-            this.Blob = blob;
-        }
+		// do not serialize
+		[JsonIgnore]
+		public CloudBlockBlob Blob { get; }
 
-        // do not serialize
-        [JsonIgnore]
-        public CloudBlockBlob Blob { get; }
+		public string Offset { get; set; }
 
-        public string Offset
-        {
-            get
-            {
-                return this.offset;
-            }
+		public long SequenceNumber { get; set; }
 
-            set
-            {
-                this.offset = value;
-            }
-        }
+		public override async Task<bool> IsExpired()
+		{
+			await this.Blob.FetchAttributesAsync().ConfigureAwait(false); // Get the latest metadata
+			var currentState = this.Blob.Properties.LeaseState;
+			return currentState != LeaseState.Leased;
+		}
 
-        public long SequenceNumber { get; set; }
+		public override async Task<string> GetStateDebug()
```

Comment:
>public override async Task<string> GetStateDebug() [](start = 2, length = 50)

Can you please remove this as part of you change? It seems some debug code leaked.